src/Iteration.cpp

```diff
@@ -174,6 +175,24 @@ Iteration::close( bool _flush )
     return *this;
 }
 
+Iteration &
+Iteration::open()
+{
+    Series * s = &auxiliary::deref_dynamic_cast< Series >(
+        parent->attributable->parent->attributable );
+    // figure out my iteration number
+    auto begin = s->indexOf( *this );
+    //auto end = begin;
+    //++end;
+    s->openIteration( begin->first, *this );
```

Comment:
Something is still wrong here, because it still hangs in the test.

Reply:
The hang is at the backends' fault i think. I've pushed a quick fix for ADIOS2, but I think ADIOS1 needs to be fixed too.

Reply:
The issue is with the backends. I've pushed a fix for ADIOS2, still need to look into the other ones